### PR TITLE
[FLINK-17465][doc-zh] Update Chinese translations for memory configuration documents.

### PR DESCRIPTION
**Pull request by xintongsong**
This PR updates Chinese translations for memory configuration documents, W.R.T. the new job manager memory configurations.

### CODE REVIEW

**Comment by xintongsong** [comment]
cc @KarmaGYZ

**Comment by KarmaGYZ** [comment]
Thanks for the PR. Regarding the links in this doc. Please take a look at the discussion [here](http://apache-flink-mailing-list-archive.1008284.n3.nabble.com/Reminder-Prefer-link-tag-in-documentation-td42362.html).

**Comment by xintongsong** [comment]
Thanks for the pointer, @KarmaGYZ.
I'm aware of this discussion. However, I would like to scope this out from this PR.
- The current links are not introduced by this translation effort, but pre-exist in the English docs. We need to modify them for the English docs anyway.
- This translation PR is already quite big and complex.

**Comment by xintongsong** [comment]
Rebased for conflict resolving, wrt 77fd975ce4e282dfeaf2ca3235b0550ffa24b36d

File: docs/ops/memory/mem_setup_tm.zh.md [hunk 1]
```diff
@@ -99,21 +91,90 @@ Flink 会根据默认值或其他配置参数自动调整剩余内存部分的
...
+
+<a name="configure-off-heap-memory-direct-or-native" />
 
 ## 配置堆外内存（直接内存或本地内存）
 
 用户代码中分配的堆外内存被归为*任务堆外内存（Task Off-heap Memory），可以通过 [`taskmanager.memory.task.off-heap.size`](../config.html#taskmanager-memory-task-off-heap-size) 指定。
 
-<span class="label label-info">提示</span> 你也可以调整[框架推外内存（Framework Off-heap Memory）](mem_detail.html#框架内存)。
+<span class="label label-info">提示</span>
+你也可以调整[框架推外内存（Framework Off-heap Memory）](#framework-memory)。
```
Comment (KarmaGYZ) [inline docs/ops/memory/mem_setup_tm.zh.md:103]:
```suggestion
你也可以调整[框架堆外内存（Framework Off-heap Memory）](#framework-memory)。
```

File: docs/ops/memory/mem_setup_tm.zh.md [hunk 2]
```diff
@@ -99,21 +91,90 @@ Flink 会根据默认值或其他配置参数自动调整剩余内存部分的
...
 这是一个进阶配置，建议仅在确定 Flink 框架需要更多的内存时调整该配置。
 
-Flink 将*框架堆外内存*和*任务堆外内存*都计算在 JVM 的*直接内存*限制中，请参考 [JVM 参数](mem_detail.html#jvm-参数)。
+Flink 将*框架堆外内存*和*任务堆外内存*都计算在 JVM 的*直接内存*限制中，请参考 [JVM 参数](mem_setup.html#jvm-parameters)。
 
-<span class="label label-info">提示</span> 本地内存（非直接内存）也可以被归在*框架堆外内存*或*任务推外内存*中，在这种情况下 JVM 的*直接内存*限制可能会高于实际需求。
+<span class="label label-info">提示</span>
+本地内存（非直接内存）也可以被归在*框架堆外内存*或*任务推外内存*中，在这种情况下 JVM 的*直接内存*限制可能会高于实际需求。
```
Comment (KarmaGYZ) [inline docs/ops/memory/mem_setup_tm.zh.md:109]:
```suggestion
本地内存（非直接内存）也可以被归在*框架堆外内存*或*任务堆外内存*中，在这种情况下 JVM 的*直接内存*限制可能会高于实际需求。
```

File: docs/ops/memory/mem_setup_tm.zh.md [hunk 3]
```diff
@@ -99,21 +91,90 @@ Flink 会根据默认值或其他配置参数自动调整剩余内存部分的
...
 若二者均未指定，会根据[默认占比](../config.html#taskmanager-memory-managed-fraction)进行计算。
 
-请同时参考[如何配置 State Backend 内存](mem_tuning.html#state-backend-的内存配置)以及[如何配置批处理作业内存](mem_tuning.html#批处理作业的内存配置)。
+请同时参考[如何配置 State Backend 内存](mem_tuning.html#configure-memory-for-state-backends)以及[如何配置批处理作业内存](mem_tuning.html#configure-memory-for-batch-jobs)。
+
+<a name="configure-off-heap-memory-direct-or-native" />
 
 ## 配置堆外内存（直接内存或本地内存）
 
 用户代码中分配的堆外内存被归为*任务堆外内存（Task Off-heap Memory），可以通过 [`taskmanager.memory.task.off-heap.size`](../config.html#taskmanager-memory-task-off-heap-size) 指定。
```
Comment (KarmaGYZ) [inline docs/ops/memory/mem_setup_tm.zh.md:100]:
```suggestion
用户代码中分配的堆外内存被归为*任务堆外内存（Task Off-heap Memory）*，可以通过 [`taskmanager.memory.task.off-heap.size`](../config.html#taskmanager-memory-task-off-heap-size) 指定。
```

**Comment by xintongsong** [comment]
Thanks @KarmaGYZ. I've fixed the typos.
Merging this now.

**Pull request closed without merging**